File: 0392-is-subsequence/0392-is-subsequence.js
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isSubsequence = function(s, t) {
  const sArray = s.split("");
  let tArray = t.split("");
  
  for (const letter of sArray) {
    const index = tArray.indexOf(letter);
    
    if (index !== -1) {
      tArray = tArray.slice(index + 1)
    } else {
      return false;
    }
  }
  
  return true;
};

File: 0392-is-subsequence/0392-is-subsequence.py
class Solution:
  def isSubsequence(self, s: str, t: str) -> bool:
    index = 0

    if s == "":
      return True

    for char in t:
      if index < len(s) and s[index] == char:
        index += 1
    
    return len(s) == index
